kolibri/utils/server.py

```diff
@@ -444,29 +459,26 @@ def stop():
     if not pid:
         return status
 
-    kill_pid(pid)
-    wait_for_status(STATUS_STOPPED)
-    if pid_exists(pid):
-        logger.debug(
-            "Process wth pid %s still exists after soft kill signal; attempting a SIGKILL."
-            % pid
-        )
-        os.kill(pid, signal.SIGKILL)
-    starttime = time.time()
-    while time.time() - starttime <= 10:
+    with open(COMMAND_FLAG, "w") as f:
+        f.write(STOP)
+    if not wait_for_status(STATUS_STOPPED, timeout=30):
         if pid_exists(pid):
-            time.sleep(0.1)
-        else:
-            break
-    if pid_exists(pid):
-        logger.debug(
-            "Process wth pid %s still exists after soft kill signal; attempting a SIGKILL."
-            % pid
-        )
-        os.kill(pid, signal.SIGKILL)
-    if pid_exists(pid):
-        logging.error("Kolibri process has failed to shutdown")
-        return STATUS_UNCLEAN_SHUTDOWN
+            logger.debug("Process wth pid %s still exists; attempting a SIGKILL." % pid)
+            try:
+                os.kill(pid, signal.SIGKILL)
```

Comment:
It looks like previously on POSIX systems, `SIGTERM` was used, but with this PR, `SIGKILL` will always be used.  Is the idea that the following lines should handle shutdown gracefully on POSIX systems, and if not a `SIGKILL` is fine?
```
self.bus.log("Stopping server.")
self.thread.cancel()
self.bus.transition("EXITED")
```

Do you see any risk to changing this for POSIX systems?

Reply:
The idea is not to use signals at all (and instead use the file based flag) for cross-OS consistency. We revert to using SIGKILL in the extreme case where the Kolibri process is unresponsive to that.